### ISSUE TITLE
QPopupEdit validation not working

### ISSUE THREAD

**Issue by y-man** [quasarframework/quasar#8416]
**Describe the bug**
I have a QPopupEdit with a QSelect inside. QSelect has a rule, that there must be at least one entry selected. But validation method on QPopupEdit is not called, if initial array is empty. After you add the first entry, save and then try to remove it again, validation is working as expected.
I did also test this with QInput, same problem.

**Codepen/jsFiddle/Codesandbox (required)**
QSelect: https://codepen.io/y-man/pen/QWGyKJZ
QInput: https://codepen.io/y-man/pen/NWbxRQO

**To Reproduce**
1. Open Codepen
2. Click on 'click here to add items'
3. Click next to the input field (you see rule of QSelect is working)
4. Click on 'set' in popup
5. Popup closes

**Expected behavior**
Validation method should be called everytime you click on 'set'.

**Platform (please complete the following information):**
MacOS 10.15.7, Node 12.20.0, Yarn 1.22.10, Chrome 88

**Comment by rstoenescu** [comment]
Hi,

The function supplied to `validate` prop should return a Boolean. In your case, when isValid is false, it returns a String...

**Comment by rstoenescu** [comment]
![popup-edit](https://user-images.githubusercontent.com/4665641/107055391-ec99d200-67d9-11eb-8c76-2da11714f126.png)

**Comment by pdanpdan** [comment]
I think it's about first time validation. I'll reopen to check it.

**Comment by pdanpdan** [comment]
@y-man The validation is only performed if you change the value


I adjusted the example just a little to fix some linting errors.

@rstoenescu It's about this part:

```js
set () {
  if (this.__hasChanged() === true) {
    if (this.validate(this.value) === false) {
      return
    }
    this.$emit('save', this.value, this.initialValue)
  }
  this.__close()
},
```

maybe we should not allow `set` if the value is not changed and is invalid

```js
set () {
  if (this.validate(this.value) !== true) {
    return
  }
  if (this.__hasChanged() === true) {
    this.$emit('save', this.value, this.initialValue)
  }
  this.__close()
},
```

**Comment by rstoenescu** [comment]
Fix will be available in quasar v1.15.3.
